### PR TITLE
Fix 309: static initializers in Log

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: lunchbox/log.cpp [hunk 1]
```diff
@@ -42,15 +42,61 @@
...
+        , LogTable(LOG_ERROR, "WARN"), LOG_TABLE_ENTRY(INFO),
+            LOG_TABLE_ENTRY(DEBUG), LOG_TABLE_ENTRY(VERB), LOG_TABLE_ENTRY(ALL)
+    }
+#ifdef NDEBUG
+    , stream(&std::cout)
+#else
+    , stream(&std::cerr)
+#endif
+          ,
+        file(nullptr), clock(&defaultClock)
```
Comment (rdumusc) [inline lunchbox/log.cpp]: style:
, file(nullptr)
, clock(&defaultClock)

File: lunchbox/log.cpp [hunk 2]
```diff
@@ -42,15 +42,61 @@
...
+#define LOG_TABLE_ENTRY(name) LogTable(LOG_##name, std::string(#name))
+
+struct LogGlobals
+{
+    LogGlobals()
+        : levels
+    {
+        LOG_TABLE_ENTRY(ERROR)
+        , LogTable(LOG_ERROR, "WARN"), LOG_TABLE_ENTRY(INFO),
+            LOG_TABLE_ENTRY(DEBUG), LOG_TABLE_ENTRY(VERB), LOG_TABLE_ENTRY(ALL)
```
Comment (rdumusc) [inline lunchbox/log.cpp]: this is not easy to read... How about removing the macro just using an initializer list:
```
LogGlobals()
    : levels{{LOG_ERROR, "ERROR"},
             {LOG_ERROR, "WARN"},
             {LOG_INFO, "INFO"},
             {LOG_DEBUG, "DEBUG"},
             {LOG_VERB, "VERB"}, 
             {LOG_ALL, "ALL"}}
```

File: lunchbox/log.cpp [hunk 3]
```diff
@@ -42,15 +42,61 @@
...
 const size_t LENGTH_PID = 5;
 const size_t LENGTH_THREAD = 8;
 const size_t LENGTH_FILE = 29;
 const size_t LENGTH_TIME = 6;
 
+namespace
+{
+struct LogTable
+{
+    LogTable(const LogLevel _level, const std::string& _name)
```
Comment (rdumusc) [inline lunchbox/log.cpp]: This constructor is probably not needed, the default constructor LogTable table{LOG_ERROR, "ERROR"} should work fine.

Comment (eile) [inline lunchbox/log.cpp]: Yes, C++03 zombie. Fixing.

File: lunchbox/log.cpp (hunk 2 above)
Comment (eile) [inline lunchbox/log.cpp]: That's a clang-format snafu. Fixed.

File: lunchbox/log.cpp (hunk 1 above)
Comment (eile) [inline lunchbox/log.cpp]: same as above.

**Comment by eile** [comment]
FIY @mgeplf

**Comment by eile** [comment]
@rdumusc : Please merge in case I don't get it done today.

File: lunchbox/log.cpp (hunk 2 above)
Comment (rdumusc) [inline lunchbox/log.cpp]: Fine like that, now it is readable though I would still have removed that macro (could also be the reason clang-format got confused).

**Comment by mgeplf** [comment]
Thanks, LGTM too

**Pull request merged**